warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexMatch.java

```diff
@@ -89,7 +89,12 @@ public boolean equals(Object obj) {
         if (null == obj)
             return false;
         if (obj instanceof IndexMatch) {
-            return uid.equals(((IndexMatch) obj).uid);
+            IndexMatch other = (IndexMatch) obj;
```

Comment:
Can we have a case where the uid is the same and yet the node set is different?

Reply:
Yes. If two distinct terms (or multiple terms in different sub-trees) hit on the same document then we have a case of two identical uids and two different JexlNodeSets.